src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,138 @@
+import { Disclosure, Transition } from '@headlessui/react'
+// @ts-ignore
+import AccordionStyles from './Accordion.module.css'
+import { IconChevronUp } from '../Icon/icons/IconChevronUp'
+import { IconContext } from '../Icon/IconContext'
+import Typography from '../Typography'
+
+type ContextValue = Required<
+  Pick<AccordionProps, 'defaultActiveId' | 'icon' | 'iconPosition' | 'onChange'>
+>
+
+const AccordionContext = createContext<ContextValue>({
+  defaultActiveId: [],
+  icon: <IconChevronUp strokeWidth={2} />,
+  iconPosition: 'right',
+  onChange: undefined,
+})
+
+interface AccordionProps {
+  children?: React.ReactNode
+  className?: string
+  defaultActiveId?: (string | number)[]
+  icon?: React.ReactNode
+  iconPosition?: 'left' | 'right'
+  bordered?: boolean
+  onChange?: (item: {
+    label: string
+    id?: string | number
+    open: boolean
+  }) => void
+}
+
+function Accordion({
+  children,
+  className,
+  defaultActiveId = [],
+  icon = <IconChevronUp strokeWidth={2} />,
+  iconPosition = 'right',
+  bordered,
+  onChange,
+}: AccordionProps) {
+  let containerClasses = [AccordionStyles['sbui-accordion-container']]
+  if (bordered) {
+    containerClasses.push(AccordionStyles['sbui-accordion-container--bordered'])
+  }
+  if (className) {
+    containerClasses.push(className)
+  }
+
+  const contextValue = {
+    defaultActiveId,
+    icon,
+    iconPosition,
+    onChange,
+  }
+
+  return (
+    <AccordionContext.Provider value={contextValue}>
+      <div className={containerClasses.join(' ')}>{children}</div>
+    </AccordionContext.Provider>
+  )
+}
+
+interface ItemProps {
+  children?: React.ReactNode
+  className?: string
+  label: string
+  id?: string | number
+}
+
+export function Item({ children, className, label, id }: ItemProps) {
+  const { defaultActiveId, icon, iconPosition, onChange } = useContext(
+    AccordionContext
+  )
+
+  let panelClasses = [AccordionStyles['sbui-accordion-item__panel']]
+
+  let buttonClasses = [AccordionStyles['sbui-accordion-item__button']]
+  if (className) {
+    buttonClasses.push(className)
+  }
+
+  const isDefaultActive = defaultActiveId.includes(id)
+
+  const handleOnChange = useCallback(
+    (open: boolean) => () => {
+      if (onChange) {
+        onChange({ id, label, open })
+      }
+    },
+    [onChange, id, label]
+  )
+
+  return (
+    <Disclosure defaultOpen={isDefaultActive}>
+      {({ open }) => (
+        <>
+          <Disclosure.Button
+            className={
+              open
+                ? `${buttonClasses.join(' ')} sbui-accordion-item__button--open`
+                : buttonClasses.join(' ')
+            }
+          >
+            <IconContext.Provider
+              value={{
+                className: open ? 'mt-px transform rotate-180' : 'mt-px',
+              }}
+            >
+              {iconPosition === 'left' && icon}
+              <Typography>{label}</Typography>
```

Comment:
Might be better to use <Typography.Text> instead here.
I feel like <Typography> doesn't work very well - as I didn't do a very good job with it.

Reply:
Cool yeah I wasn't quite sure what to use, maybe I shall have a little poke around at Typography next